lib/decimal.ex

```diff
@@ -1237,6 +1237,32 @@ defmodule Decimal do
     |> new()
   end
 
+  @doc """
+  Creates a new decimal number from an integer, string or float.
+
+  Because conversion from a floating point number is not exact, it's recommended
+  to instead use `new/1` or `from_float/1` when the argument's type is certain.
+  See `from_float/1`.
+
+  ## Examples
+
+      iex> Decimal.from_any(3)
+      #Decimal<3>
+
+      iex> Decimal.from_any(3.0)
+      #Decimal<3.0>
+
+      iex> Decimal.from_any("3")
+      #Decimal<3>
+
+      iex> Decimal.from_any("3.0")
+      #Decimal<3.0>
+
+  """
+  @spec from_any(number | binary) :: t
+  def from_any(float) when is_float(float), do: from_float(float)
+  def from_any(value), do: new(value)
```

Comment:
Thanks for the PR!

because we use Decimal.new under the hood, this will work too:

```elixir
iex> Decimal.from_any(Decimal.new(42))
#Decimal<42>
```

Could you send another patch with an update to docs and specs to mention we accept decimals here too? Theres a `decimal :: t | integer | String.t` type that you can use.

Reply:
Thanks, this will be a big help for us. And I'd be happy to submit that update as well, good call!